packages/block-library/src/navigation/edit/index.js

```diff
@@ -152,13 +165,22 @@ function Navigation( {
 				( block ) => block.name === 'core/navigation-submenu'
 			);
 
+			const { canUser, hasFinishedResolution } = select( coreStore );
+
 			return {
 				hasSubmenus: firstSubmenu,
 				innerBlocks: blocks,
 				isInnerBlockSelected: hasSelectedInnerBlock( clientId, true ),
+				canUserPublishNavigation: ref
+					? canUser( 'publish', 'navigation', ref )
+					: undefined,
+				hasResolvedCanUserPublishNavigationEntity: hasFinishedResolution(
+					'canUser',
+					[ 'publish', 'navigation', ref ]
+				),
```

Comment:
Like the feedback on the other PR, I think this should be folded into `useNavigationMenu`.

We need to avoid making the navigation block main edit file any chunkier.